pkg/envoy/boilerplate.go

```diff
@@ -61,6 +62,10 @@ func init() {
 
 func makeVirtualHost(vhost *virtualHost, reselectionAttempts int64) *route.VirtualHost {
 	routes := []*route.Route{}
+	// sort vhosts routes to make sure the bigger matching url is first
+	sort.Slice(vhost.Routes, func(i, j int) bool {
+		return len(vhost.Routes[i].Route.String()) > len(vhost.Routes[j].Route.String())
```

Comment:
Couple questions come to mind:
1. Is the sorting consistent for routes of equal length, or will it be random which of such ends up being first?
2. Do routes allow wildcards in them? If yes, sorting by route length doesn't sound good to me, as (for a very simple example) eg. `/.*` would then match before `/a` when we obviously would like the latter to be selected, being the more specific route

Reply:
for first question if route length is equal then the sort keep order base on actual list value[ here an example](https://go.dev/play/p/Itv68dzVlLF) 

we can have different type of sorting base of configuration 
for example `regex` order which prefer regex and "/" to sort least